src/diffusers/image_processor.py

```diff
@@ -498,9 +514,20 @@ def preprocess(
                 else:
                     image = np.expand_dims(image, axis=-1)
 
-        if isinstance(image, supported_formats):
+        # image processor only accepts image or a list of images or a batch of images (4d array/tensors) as inputs,
```

Comment:
I guess it's slightly confusing to differentiate between "list of images" and "batch of images". Maybe better to also provide the type-annotation in the comments to avoid any confusion?